src/actions/sso/index.ts

```diff
@@ -29,5 +50,59 @@ export const consumeCredentialRequest = (jwtEncodedCR: string) => {
         credentials: attributeSummaries
       }
     }))
+
+    const summary = {
+      requester: CR.getRequester(),
+      callbackURL: CR.getCallbackURL(),
+      request: credentialRequests
+    }
+
+    dispatch(setCredentialRequest(summary))
+    dispatch(navigationActions.navigate({routeName: routeList.Consent}))
+  }
+}
+
+// TODO Decrypt when fetching from storage
+export const sendCredentialResponse = (selectedCredentials: StateVerificationSummary[]) => {
+  return async(dispatch: Dispatch<AnyAction>, getState: Function, backendMiddleware: BackendMiddleware) => {
+    const { jolocomLib, storageLib, keyChainLib, encryptionLib, ethereumLib } = backendMiddleware
+
+    const encryptionPass = await keyChainLib.getPassword()
```

Comment:
I use the same logic for storage - we should move it to a function that takes backend middleware.